pkg/util/session_pool.go

```diff
@@ -29,6 +29,13 @@ type SessionPool interface {
 	Close()
 }
 
+// DestroyableSessionPool is a session pool that can destroy the session resource.
+// If the caller meets an error when using the session, it can destroy the session.
+type DestroyableSessionPool interface {
```

Comment:
How about adding "StoreInternalSession" and "DeleteInternalSession" as an exampling here? Otherwise it's hard to understand why we have to manually destroy sessions if error...

Reply:
Added.